web/client/plugins/Map.jsx

```diff
@@ -281,7 +282,7 @@ class MapPlugin extends React.Component {
         const projection = this.props.map.projection || 'EPSG:3857';
         return [...this.props.layers, ...this.props.additionalLayers].filter(this.filterLayer).map((layer, index) => {
             return (
-                <plugins.Layer type={layer.type} srs={projection} position={index} key={layer.id || layer.name} options={layer} securityToken={this.props.securityToken}>
+                <plugins.Layer type={layer.type} srs={projection} position={index} key={layer.id || layer.name || uuidv1()} options={layer} securityToken={this.props.securityToken}>
```

Comment:
What the use case for a layer without both id and name? Do you have an example?

Reply:
when you use a vector layer in the additionalLayers list

Reply:
I suggest to add unique id to every additional layer you create it.  I suggest to assign one on creation. If missing, please use a default one in reducer, @mbarto do you agree?

Reply:
@offtherailz i'm just passing a random value if it does not exist to the **key** prop to avoid have a warning in the console

Reply:
The warning, in that case, is correct. If some layer has id or name missing, probably you're doing something wrong, and it is better to be notified of this.